### PR TITLE
Make kubejob usable as library

### PR DESCRIPTION
**Pull request by mwuertinger**
This change pulls the main functionality of the tool out into an easy to use function so that other applications can benefit from it.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -34,12 +28,27 @@ func main() {
...
-	jobIn, err := parseAndValidateJob(args.JobFile)
+	go func() {
+		sigChan := make(chan os.Signal)
+		signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+		// wait for signal
+		<-sigChan
+
+		// cancel context
+		cancel()
```
Comment (dan-force) [inline main.go]: Is that the same cancel as in line 38?
What happens if `defer` tries to cancel that context again?

File: main.go [hunk 2]
```diff
@@ -49,41 +58,50 @@ func main() {
...
+			case job.LogLine:
+				log.Printf("%s: %s", event.Container, event.Line)
+			case core.PodStatus:
+				status := event
+				if status.Phase != lastPhase {
+					log.Print("Phase: ", status.Phase)
+				}
+				lastPhase = status.Phase
+
+				if status.Phase == core.PodPending {
```
Comment (dan-force) [inline main.go]: Minor: You could reverse the logic and `continue` if `status.Phase != core.PodPending`. Saves some nesting.

File: main.go [hunk 3]
```diff
@@ -49,41 +58,50 @@ func main() {
...
 
-	// create the job
-	job, err := cs.BatchV1().Jobs(args.Namespace).Create(jobIn)
-	if err != nil {
-		log.Fatal("Failed to create job: ", err)
+	success, err := job.RunJob(ctx, cs, args.Namespace, jobSpec, events)
+	if success {
+		log.Print("Job completed successfully")
+	} else {
+		log.Print("Job failed")
```
Comment (dan-force) [inline main.go]: Maybe add the actual error to the log output?

File: main.go (hunk 1 above)
Comment (mwuertinger) [reply main.go]: https://golang.org/pkg/context/#CancelFunc
"After the first call, subsequent calls to a CancelFunc do nothing."

**Pull request merged**